web/src/main/java/no/nav/foreldrepenger/web/app/tjenester/fpoversikt/FpDtoTjeneste.java

```diff
@@ -192,10 +196,28 @@ private boolean morArbeidUtenDok(Behandling behandling) {
         if (søknad.isEmpty() || mottattEllerVenterDokumentasjonPåAtMorErIArbeid(søknad.get())) {
             return false;
         }
-        var søknadsperioder = ytelseFordelingTjeneste.hentAggregatHvisEksisterer(behandling.getId())
-            .map(yfa -> yfa.getOppgittFordeling().getPerioder())
-            .orElse(List.of());
-        return søknadsperioder.stream().anyMatch(OppgittPeriodeEntitet::erAktivitetskravMedMorArbeid);
+        var ytelseFordelingAggregat = ytelseFordelingTjeneste.hentAggregatHvisEksisterer(behandling.getId());
+        if (ytelseFordelingAggregat.isEmpty()) {
+            return false;
+        }
+        var søknadsperioder = ytelseFordelingAggregat.get().getOppgittFordeling().getPerioder();
+        return søknadsperioder.stream().anyMatch(
+            oppgittPeriodeEntitet -> harAktivitetskravMorArbeid(oppgittPeriodeEntitet, ytelseFordelingAggregat.get().getOppgittRettighet()));
+    }
+
+    private static boolean harAktivitetskravMorArbeid(OppgittPeriodeEntitet oppgittPeriodeEntitet, OppgittRettighetEntitet oppgittRettighet) {
+        if (!Objects.equals(oppgittPeriodeEntitet.getMorsAktivitet(), no.nav.foreldrepenger.behandlingslager.behandling.ytelsefordeling.MorsAktivitet.ARBEID)) {
+            return false;
+        }
+        if (Objects.equals(UttakPeriodeType.FELLESPERIODE, oppgittPeriodeEntitet.getPeriodeType())) {
+            return true;
+        }
+        return oppgittBareSøkerRett(oppgittRettighet);
+    }
+
+    private static boolean oppgittBareSøkerRett(OppgittRettighetEntitet oppgittRettighet) {
```

Comment:
Har du ikke noe utilitymetoder på aggregatet for dette? Evt OmsorgUtil + aggregat.
Så er det overstyrt rettighet å ta stilling til - som kan implisere noe annet enn søknaden.
Spørsmålet er kanskje mer om det ikke holder å sjekke på FELLESPERIODE - MorsAktivitet skal ikke være oppgitt for aleneomsorg eller bareMor. Dessuten ser vel utledningen også på rolleType.ikkeMor?